### PR TITLE
Add api delete by id to operation

### PR DESCRIPTION
**Pull request by Incarnation-p-lee**

### CODE REVIEW

File: src/main/java/com/microsoft/spring/data/gremlin/conversion/script/GremlinScriptLiteral.java [hunk 1]
```diff
@@ -19,20 +19,23 @@
...
-
     /**
      * Generate the deleteAll query from source (Vertex, Edge or Graph).
      */
     List<String> generateDeleteAllScript(GremlinSource source);
 
-
     /**
-     * Generate the findById query from source (Vertex, Edge or Graph).
+     * Generate the findById query from source (Vertex, Edge).
```
Comment (sophiaso) [inline src/main/java/com/microsoft/spring/data/gremlin/conversion/script/GremlinScriptLiteral.java]: Why this `or Graph` is removed in the comment, but some other functions below are kept?

File: src/main/java/com/microsoft/spring/data/gremlin/conversion/script/GremlinScriptLiteralEdge.java [hunk 2]
```diff
@@ -108,5 +108,25 @@
...
+
+        Assert.notNull(id, "id should not be null");
+
+        scriptList.add(Constants.GREMLIN_PRIMITIVE_GRAPH);
+        scriptList.add(String.format(Constants.GREMLIN_PRIMITIVE_EDGE, id));
+        scriptList.add(Constants.GREMLIN_PRIMITIVE_DROP);
+
+        final String query = String.join(Constants.GREMLIN_PRIMITIVE_INVOKE, scriptList);
+
+        return Arrays.asList(query);
```
Comment (sophiaso) [inline src/main/java/com/microsoft/spring/data/gremlin/conversion/script/GremlinScriptLiteralEdge.java]: Seems all the script is a single `String`, but all the methods in this class are wrapping the `single String` into a `String list`, any reason why not using `String` as return type here? Just curious.

File: src/main/java/com/microsoft/spring/data/gremlin/conversion/script/GremlinScriptLiteral.java (hunk 1 above)
Comment (Incarnation-p-lee) [reply src/main/java/com/microsoft/spring/data/gremlin/conversion/script/GremlinScriptLiteral.java]: From the gremlin server side, we cannot find the graph by Id. As the domains in a graph will lose the type. We cannot retrieve that for now. The comments is not verify clear before, fix it here.

File: src/test/java/com/microsoft/spring/data/gremlin/query/GremlinTemplateIT.java [hunk 3]
```diff
@@ -357,5 +357,51 @@ public void testSaveEdge() {
...
     }
+
+    @Test
+    public void testVertexDeleteById() {
+        this.template.deleteById(this.person.getId(), Person.class);
+        this.template.insert(this.person);
+        this.template.deleteById(this.person0.getId(), Person.class);
+
+        Person foundPerson = this.template.findById(this.person.getId(), Person.class);
+        Assert.assertNotNull(foundPerson);
```
Comment (sophiaso) [inline src/test/java/com/microsoft/spring/data/gremlin/query/GremlinTemplateIT.java]: Add any assertion failure message here?

File: src/main/java/com/microsoft/spring/data/gremlin/conversion/script/GremlinScriptLiteralEdge.java (hunk 2 above)
Comment (Incarnation-p-lee) [reply src/main/java/com/microsoft/spring/data/gremlin/conversion/script/GremlinScriptLiteralEdge.java]: We can consider that all query is compose of string list. In this way, we can avoiding query string handling mess like joint/contract/replace. Just take care of its own.

File: src/test/java/com/microsoft/spring/data/gremlin/query/GremlinTemplateIT.java (hunk 3 above)
Comment (Incarnation-p-lee) [reply src/test/java/com/microsoft/spring/data/gremlin/query/GremlinTemplateIT.java]: It is UT. I think it is clean enough here.

**Pull request merged**